packages/app/components/ConnectButton.tsx

```diff
@@ -0,0 +1,94 @@
+"use client";
+
+import { ConnectKitButton } from "connectkit";
+import { BodyText, Button } from "../ui";
+import { useEnsAvatar, useBalance } from "wagmi";
+import Image from "next/image";
+
+const CustomConnectButton = ({
+  address,
+  ensName,
+  onClick,
+}: {
+  address: `0x${string}`;
+  onClick: () => void;
+  ensName?: string;
+}) => {
+  const { data: avatar } = useEnsAvatar({
+    name: ensName,
+    chainId: 1,
+  });
+  const { data: balance } = useBalance({
+    address: address,
+  });
+
+  const truncatedAddress = (size: number) =>
+    `${address.slice(0, size)}...${address.slice(
+      address.length - size,
+      address.length
+    )}`;
+
+  const formattedBalance = (balanceData: NonNullable<typeof balance>) =>
```

Comment:
doesn't `viem` has formating helpers?

Reply:
Didn't find so custom formatting. Will try to find something, otherwise will stick with this solution for now.

Reply:
There isn't anything to format the balance like we need.